Week4/week 4.ipynb

```diff
@@ -0,0 +1,335 @@
+    "        (\\\n",
+    "        fname text,\\\n",
+    "        lname text,\\\n",
+    "        contact integer,\\\n",
+    "        datetime text,\\\n",
+    "        operation text\\\n",
+    "        );\"\n",
+    "    curobj.execute(logtbl)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "079f160a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " def contact_validate_trigger():\n",
+    "    con_validate_trigger=\"create trigger IF NOT EXISTS contact_validate\\\n",
+    "                        before insert\\\n",
+    "                        on contact\\\n",
+    "                        begin\\\n",
+    "                        select\\\n",
+    "                                case \\\n",
+    "                                when length(new.contact)>10 then\\\n",
+    "                                raise(abort,'Given number is greater than 10 digit . Please enter 10 digit number')\\\n",
+    "                                when length(new.contact)<10 then\\\n",
+    "                                raise(abort,'Given number is less than 10 digit . Please enter 10 digit number')\\\n",
+    "                        end;\\\n",
+    "                        end;\"\n",
+    "\n",
+    "    curobj.execute(con_validate_trigger)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "5dafecb2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def log_table_trigger():\n",
+    "    log_tbl_insert_trigger=\"create trigger IF NOT EXISTS insert_trigger\\\n",
+    "                        after insert \\\n",
+    "                        on contact\\\n",
+    "                        begin\\\n",
+    "                            insert into logtable\\\n",
+    "                            values(new.fname,new.lname,new.contact,datetime('now','localtime'),'INSERT');\\\n",
+    "                        end;\"\n",
+    "\n",
+    "    curobj.execute(log_tbl_insert_trigger)\n",
+    "    log_table_delete_trigger=\"create trigger IF NOT EXISTS delete_trigger\\\n",
+    "                                after delete \\\n",
+    "                                on contact\\\n",
+    "                                begin\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(old.fname,old.lname,old.contact,datetime('now','localtime'),'DELETE');\\\n",
+    "                                end;\"\n",
+    "\n",
+    "    curobj.execute(log_table_delete_trigger)\n",
+    "    log_table_update_trigger=\"create trigger IF NOT EXISTS update_trigger\\\n",
+    "                                after update\\\n",
+    "                                on contact\\\n",
+    "                                begin\\\n",
+    "                                insert into logtable\\\n",
+    "                                values(new.fname,new.lname,new.contact,datetime('now','localtime'),'UPDATE(After)');\\\n",
```

Comment:
whats datetime()?

Reply:
Datetime function in trigger is used to record current date and time like at what date and time the record is inserted by user .